1175: Language switch

f1sh1918 opened the pull request:
### Short description

Since translations are available, the user want to switch the language in the app

### Proposed changes

- added a settings section with a language change
- adjusted the tabItem for sozialpass

### Side effects

- the title of the content page header is not affected when the language was changed. Everything else works properly. No clue why this is happening .

### Resolved issues

<!-- List all issues which should be closed when this PR is merged. -->

Fixes: #1075

f1sh1918 commented:
Maybe @sarahsporck or @michael-markl have an idea why only this title "Sprache wechseln" is not changing.
Didn't find anything that this `SliverAppBar` is special

<img src="https://github.com/digitalfabrik/entitlementcard/assets/37902063/bc94ac48-aff6-4647-b85d-4f1eae90834a"  width="300" height="600" />

steffenkleinle reviewed (commented):
Tested on android, works nicely!
One thing I noticed (not introduced in your PR though): If you click on `Change language`, the button does not show the touch. Once navigating back to the about screen/closing the language switch, the button then shows the touch.

f1sh1918 commented on frontend/build-configs/bayern/index.ts:
```diff
@@ -62,7 +62,8 @@ export const bayernCommon: CommonBuildConfigType = {
     featureFlags: {
-        verification: true
+        verification: true,
+        settings: false
```
Thats a good point. Gonna use `appLocales` 
If we have more settings options we can do it more granulary

f1sh1918 commented on frontend/lib/about/about_page.dart:
```diff
@@ -95,53 +98,76 @@ class AboutPageState extends State<AboutPage> {
-                  builder: (context) =>
-                      SimpleDialog(title: Text(t.about.developmentOptions), children: [DevSettingsView()]),
-                ),
+                leading: const Icon(Icons.privacy_tip_outlined),
```
i just move everything except of dev_settings a `section`. I just changed the structure

f1sh1918 commented on frontend/assets/l10n/app_de.json:
```diff
@@ -15,6 +15,11 @@
     },
     "privacyDeclaration": "Datenschutzerklärung",
     "publisher": "Herausgeber",
+    "settings": {
+      "headline": "Einstellungen",
+      "languageChange": "Sprache wechseln",
+      "languageNotification": "Ihre Sprache wurde erfolgreich geändert!"
+    },
```
i applied first suggestions.

I would keep the second one for the following reason.
We have under about 2 sections (info, and settings)
Each one should have a separate namespace in my opinion (check last commit)

steffenkleinle replied:
I think I disagree. Imo there are two options:
- Either the sections (settings and info) are unrelated to the about namespace and therefore could also be implemented as two standalone screen: Then they should have their own namespace and not be a subnamespace (i.e. the translations are "settings" or "info" specific)
- Or the sections are just an implementation detail and should therefore not be represented here.

>  Each one should have a separate namespace in my opinion (check last commit)

Why does an own namespace make sense in your opinion? I think it just complicates things/reduces reusability without adding anymore relevant information.

f1sh1918 replied:
ok i'm fine without namespaces. I really don't care. I found the structure easier to read

steffenkleinle replied:
Sorry, I didn't mean to enforce anything here. I am fine with your solution as well!

f1sh1918 commented:
> Tested on android, works nicely! One thing I noticed (not introduced in your PR though): If you click on `Change language`, the button does not show the touch. Once navigating back to the about screen/closing the language switch, the button then shows the touch.

Hm i think i don't understand what you mean..

steffenkleinle commented:
> > Tested on android, works nicely! One thing I noticed (not introduced in your PR though): If you click on `Change language`, the button does not show the touch. Once navigating back to the about screen/closing the language switch, the button then shows the touch.
> 
> Hm i think i don't understand what you mean.. Updated pr with your suggestions and changed the identity logo as it was changed in design

See attached video. The "onTouch" animation is only shown when coming back from the language switch screen, not directly on touch. Perhaps thats also intended though.

https://github.com/digitalfabrik/entitlementcard/assets/29095487/30bcc9dc-ea16-4391-a94b-ee9a9478d021

f1sh1918 commented:
> > > Tested on android, works nicely! One thing I noticed (not introduced in your PR though): If you click on `Change language`, the button does not show the touch. Once navigating back to the about screen/closing the language switch, the button then shows the touch.
> > 
> > 
> > Hm i think i don't understand what you mean.. Updated pr with your suggestions and changed the identity logo as it was changed in design
> 
> See attached video. The "onTouch" animation is only shown when coming back from the language switch screen, not directly on touch. Perhaps thats also intended though.
> 
>  screen-20231108-151321.mp4

okay thats weird. you can create an issue for that. don't think that relies to my pr

steffenkleinle commented:
> > > > Tested on android, works nicely! One thing I noticed (not introduced in your PR though): If you click on `Change language`, the button does not show the touch. Once navigating back to the about screen/closing the language switch, the button then shows the touch.
> > > 
> > > 
> > > Hm i think i don't understand what you mean.. Updated pr with your suggestions and changed the identity logo as it was changed in design
> > 
> > 
> > See attached video. The "onTouch" animation is only shown when coming back from the language switch screen, not directly on touch. Perhaps thats also intended though.
> > screen-20231108-151321.mp4
> 
> okay thats weird. you can create an issue for that. don't think that relies to my pr

Sure, I can do that. Can you reproduce that?

f1sh1918 commented:
> > > > > Tested on android, works nicely! One thing I noticed (not introduced in your PR though): If you click on `Change language`, the button does not show the touch. Once navigating back to the about screen/closing the language switch, the button then shows the touch.
> > > > 
> > > > 
> > > > Hm i think i don't understand what you mean.. Updated pr with your suggestions and changed the identity logo as it was changed in design
> > > 
> > > 
> > > See attached video. The "onTouch" animation is only shown when coming back from the language switch screen, not directly on touch. Perhaps thats also intended though.
> > > screen-20231108-151321.mp4
> > 
> > 
> > okay thats weird. you can create an issue for that. don't think that relies to my pr
> 
> Sure, I can do that. Can you reproduce that?

yes that seem to happen for all `ContentTile` components only on android
